File: services/frontend/src/domains/equipment.ts
import type { DomainConfig, FieldConfig } from './base'
import type { GeneratedEquipmentAttrs } from './generated'
import { GENERATED } from './generated'

export type EquipmentAttrs = GeneratedEquipmentAttrs

const FIELDS: FieldConfig<EquipmentAttrs>[] = [
  { key: 'make', label: 'Make', kind: 'text' },
  { key: 'model', label: 'Model', kind: 'text' },
  { key: 'serial_number', label: 'Serial number', kind: 'text' },
  { key: 'purchase_date', label: 'Purchase date', kind: 'date' },
]

export const equipmentConfig: DomainConfig<EquipmentAttrs> = {
  label: 'Equipment',
  statuses: GENERATED.equipment.statuses,
  logTypes: GENERATED.equipment.logTypes,
  fields: FIELDS,
  defaultAttributes: () => ({ domain: 'equipment' }),
  namePlaceholder: 'Display name',
  locationPlaceholder: 'Garage, kitchen, ...',
  uiVariant: 'schedule',
}

File: libs/shared/src/aria_shared/models/entities/equipment.py
from datetime import date
from typing import ClassVar, Literal

from .base import BaseAttributes


class EquipmentAttrs(BaseAttributes):
    DOMAIN: ClassVar[str] = "equipment"
    VALID_STATUSES: ClassVar[set[str]] = {"active", "in_service", "retired"}
    LOG_TYPES: ClassVar[set[str]] = {"service", "repair", "inspection", "expense", "note", "milestone"}

    domain: Literal["equipment"] = "equipment"
    make: str | None = None
    model: str | None = None
    serial_number: str | None = None
    purchase_date: date | None = None
